Sources/SwiftFusion/Datasets/G2OReader.swift

```diff
@@ -47,25 +47,42 @@ public enum G2OReader {
   }
 
   /// A G2O problem expressed as a `NewFactorGraph`.
-  public struct G2ONewFactorGraph {
+  public struct G2ONewFactorGraph<Pose: LieGroup> {
     /// The initial guess.
     public var initialGuess = VariableAssignments()
 
+    public var initialGuessId = Array<TypedID<Pose, Int>>()
```

Comment:
I suggest naming this something that refers to "variable id" instead of "initial guess" because identifies variables generally not just initial guesses. Like: `poseVariableID`

Reply:
Agreed :) Done.